routers/account.py

```diff
@@ -273,21 +350,25 @@ def _render_run(pr: PublishedRun):
 
 def api_keys_tab(request: Request):
     gui.write("# 🔐 API Keys")
-    manage_api_keys(request.user)
+    workspace = get_current_workspace(request.user, request.session)
+    manage_api_keys(workspace, request.user)
 
 
 @contextmanager
-def account_page_wrapper(request: Request, current_tab: TabData):
+def account_page_wrapper(request: Request, current_tab: AccountTabs):
     if not request.user or request.user.is_anonymous:
         next_url = request.query_params.get("next", "/account/")
         redirect_url = furl("/login", query_params={"next": next_url})
         raise gui.RedirectException(str(redirect_url))
 
-    with page_wrapper(request):
+    with page_wrapper(request, route_fn=current_tab.route):
```

Comment:
pass down `current_tab` and build the redirect url from the tab. Passing the route_fn is very confusing

Reply:
thank you, I particularly wanted review here.

my reasoning here to choose route was that tabs `AccountTabs` are scoped down to only the account page, whereas we use routes everywhere.
`page_wrapper` is used for all routes on the app, and we could scale this to other routes besides `AccountTabs`.

Reply:
Lets not do that premature optimization and keep things simple. You're using the `route_fn` to do redirects based on some state, and those should really be inside the routes themselves. The usage for route_fn everywhere else is only to reverse URLs, something django for eg does via a special name param on every route (and we use the route's fn name as a proxy for that)

Reply:
That makes sense